.github/workflows/Syntax-Check.yml

```diff
@@ -0,0 +1,22 @@
+name: PHP Syntax Check
+
+on: [push, pull_request]
+
+jobs:
+  syntax-checker:
+    runs-on: ${{ matrix.operating-system }}
+    strategy:
+      matrix:
+        operating-system: [ ubuntu-latest ]
+        php: [ '5.4', '5.5', '5.6', '7.0', '7.1', '7.2', '7.3', '7.4', '8.0' ]
```

Comment:
Is this large matrix really necessary? Sooo many wasted resources on redundant testing.

Reply:
well this the reality of smf,
as dev you should had all version local to test your stuff.

So from my pov, is this the least point where the stuff is important to change.

Reply:
It just does a syntax check for all supported php versions.  This should help prevent us from making a mistake of using a invalid syntax in a PHP version we don't have on our dev environments.  We could still introduce those, but less likely to be fatal hopefully.